server/task.go

```diff
@@ -901,6 +905,10 @@ func areAllTasksDone(pipeline *api.Pipeline) bool {
 func (s *Server) cancelDependingTasks(ctx context.Context, task *api.Task) error {
 	queue := []int{task.ID}
 	seen := map[int]bool{task.ID: true}
+	taskStatusPatch := api.TaskStatusPatch{
```

Comment:
Instead of initializing taskStatusPatch here, let's make it inline right before we call PatchTaskStatus().